tests/integration/test_symlink.py

```diff
@@ -0,0 +1,137 @@
+# vim:ts=4:sw=4:et:
+# Copyright 2016-present Facebook, Inc.
+# Licensed under the Apache License, Version 2.0
+import WatchmanTestCase
+import tempfile
+import os
+import os.path
+import sys
+import unittest
+
+
+class TestSymlink(WatchmanTestCase.WatchmanTestCase):
+
+    # test to see that valid symbolic link
+    # updates are picked up by the symlink_target field
+    @unittest.skipIf(os.name == 'nt', 'win')
+    def test_symlink(self):
+        root = self.mkdtemp()
+        self.touchRelative(root, '222')
+        self.touchRelative(root, '333')
+        os.symlink(os.path.join(root, '222'), os.path.join(root, '111'))
+
+        self.watchmanCommand('watch', root)
+        self.assertFileList(root, ['111', '222', '333'])
+
+        expr = {
+            "expression": ["name", "111"],
+            "fields": ["name", "symlink_target"]
+        }
+
+        res1 = self.watchmanCommand('query', root, expr)
+
+        files1 = res1['files']
+        self.assertTrue(files1 and os.path.basename(files1[0]['symlink_target']) == '222')
```

Comment:
prefer to do:

```
self.assertEquals('222', os.path.basename(files1[0]['symlink_target']))
```

so that the test assertion will show what the actual value was if it ever fails